src/main/java/net/runelite/rsb/wrappers/RSModel.java

```diff
@@ -316,32 +305,28 @@ public Point getCenterPoint() {
 		return new Point((max_x + min_x) / 2, (max_y + min_y) / 2);
 	}
 
+	/**
+	 * This function first chooses a random triangle from the model polygons.
+	 * Then it generates two random numbers between 0 and 1.
+	 * These numbers are used as barycentric coordinates to generate
+	 * a point that is guaranteed to be within the chosen triangle.
+	 *
+	 * @return a random point that collides with the polygons of this model.
+	 */
 	public Point getPointNearCenter() {
 		Polygon[] triangles = this.getTriangles();
-		int min_x = Integer.MAX_VALUE, max_x = Integer.MIN_VALUE, min_y = Integer.MAX_VALUE, max_y = Integer.MIN_VALUE;
 
-		for (Polygon triangle : triangles) {
-			for (int i = 0; i < triangle.npoints; ++i) {
-				if (triangle.xpoints[i] < min_x) {
-					min_x = triangle.xpoints[i];
-				}
-				if (triangle.xpoints[i] > max_x) {
-					max_x = triangle.xpoints[i];
-				}
-				if (triangle.ypoints[i] < min_y) {
-					min_y = triangle.ypoints[i];
-				}
-				if (triangle.ypoints[i] > max_y) {
-					max_y = triangle.ypoints[i];
-				}
-			}
-		}
+		// pick a random triangle
+		int randomIndex = StdRandom.uniform(triangles.length);
+		Polygon triangle = triangles[randomIndex];
 
-		int centerX = (max_x + min_x) / 2;
-		int centerY = (max_y + min_y) / 2;
+		// pick random barycentric coordinates
+		double r1 = StdRandom.uniform();
+		double r2 = StdRandom.uniform();
 
-		int x = (int)StdRandom.gaussian(min_x, max_x, centerX, (max_x - min_x) / 3);
-		int y = (int)StdRandom.gaussian(min_y, max_y, centerY, (max_y - min_y) / 3);
+		// generate a point within the triangle using the barycentric coordinates
+		int x = (int) ((1 - Math.sqrt(r1)) * triangle.xpoints[0] + (Math.sqrt(r1) * (1 - r2)) * triangle.xpoints[1] + (Math.sqrt(r1) * r2) * triangle.xpoints[2]);
+		int y = (int) ((1 - Math.sqrt(r1)) * triangle.ypoints[0] + (Math.sqrt(r1) * (1 - r2)) * triangle.ypoints[1] + (Math.sqrt(r1) * r2) * triangle.ypoints[2]);
 
 		return new Point(x, y);
```

Comment:
This is similar to how it worked before. The problem is because of the screen projection the majority of triangles are going to be at or near the edge of the model in 2d screen space. Couple this together with the fact the model or camera may be moving and the mouse takes time to move. This causes moving the mouse to a model to fail most of the time at least when the model or camera is moving at all. I think that we can actually get the click hitbox somehow and take the center point of that which makes more sense but I didn't spend the time to figure it out. Ideally, our mouse movement would update while moving so it always finished on the model but that isn't how it works currently. Regardless, I don't think this is a good change even though on the surface it seems like a better way to do it. Oh also the name is definitely not apt with this change.

Reply:
When interacting with a furnace the old code had a 10% fail rate when zoomed in. An 80% fail rate when zoomed out.

Reply:
I've updated the interaction logic